labelbox/schema/dataset.py

```diff
@@ -260,11 +260,10 @@ def upload_if_necessary(item):
             row_data = item['row_data']
             if os.path.exists(row_data):
                 item_url = self.client.upload_file(item['row_data'])
-                item = {
-                    "row_data": item_url,
-                    "external_id": item.get('external_id', item['row_data']),
-                    "attachments": item.get('attachments', [])
-                }
+                item['row_data'] = item_url
+                if 'external_id' not in item:
```

Comment:
**question** im a little confused on how this resolved the issue. from the previous code, wouldn't it also do the same?

`"external_id": item.get('external_id', item['row_data']),`

Reply:
resolving convo as this was asked in another comment